architecture/adr-list/adr-002-identity-transactions.md

```diff
@@ -0,0 +1,272 @@
+
+_**Note**: Hyperledger Indy also contains other transaction types beyond the ones listed above, but these are currently not in scope for implementation in `cheqd-node`. They will be considered for inclusion later in the product roadmap._
+
+#### Dropping `ATTRIB` transactions
+
+`ATTRIB` was originally used in Hyperledger Indy to add document content similar to DID Documents (DIDDocs). The cheqd DID method replaces this by implementing DIDDocs for most transaction types.
+
+## Decision
+
+### cheqd DID Method
+
+The `did:cheqd` method DID has four components that are concatenated to make a W3C DID specification conformant identifier. The components are:
+
+- **DID**: the hardcoded string `did:` to indicate the identifier is a DID
+- **`cheqd` DID method**: the hardcoded string `cheqd:` indicating that the identifier uses the `cheqd` DID method specification.
+- **Namespace**: A string that identifies the name of the primary cheqd ledger ("mainnet"), followed by a `:`. The namespace string may optionally have a secondary ledger name prefixed by a `:` following the primary name. If there is no secondary ledger element, the DID resides on the primary ledger ("mainnet"), else it resides on the secondary ledger. By convention, the primary is a production ledger while the secondary ledgers are non-production ledgers (e.g. testnet) associated with the primary ledger.
+- **Namespace Identifier**: A self-certified identifier unique to the given cheqd DID ledger namespace. To be self-certifying, the identifier must be derived from the initial `verkey` associated with the identifier.
+
+The components are assembled as follows:
+
+`did:cheqd:<namespace>:<namespace identifier>`
+
+Some examples of `did:cheqd` method identifiers are:
+
+- A DID written to the cheqd mainnet ledger: `did:cheqd:mainnet:7Tqg6BwSSWapxgUDm9KKgg`
+- A DID written to the cheqd testnet ledger: `did:cheqd:testnet:6cgbu8ZPoWTnR5Rv5JcSMB`
+
+### General structure of transaction requests
+
+All identity requests will have the following format:
+
+```javascript
+{
+    "data": { "<request data for writing a transaction to the ledger>" },
+    "creators": ["<identifier>", "..."],
+    "signatures": [
+        "<public_key>": "<signature>",
+      ],
+    "metadata": {}
+}
+```
+
+- **`data`**: Data requested to be written to the ledger, specific for each request type.
+- **`creators`**: Creators identifiers (DID) list for this entity. There should be a new DIDs or an existing DIDs, for existing entities.
+- **`signatures`**: `data` should be signed by all `creators` private key. This field contains a dict there creator's public key is a key, and the signature is a value.
+- **`metadata`**: Dictionary with additional metadata fields. Empty for now. This fields provides extensibility in the future, e.g., it can contain `protocolVersion` or other relevant metadata associated with a request.
```

Comment:
It's not a DIDDoc metadata. It's a write request metadata for every request.

Reply:
I didn't add `didDocumentMetadata` and `didResolutionMetadata` because I didn't find cases then they are needed. And  `deactivated` field. I think this field is superfluous, since we don't have such a business requirement. And according to the experience of working on entities that act for a certain amount of time (TAA, revocations), this raises many legal questions for business and non-trivial nuances of implementation.
Please correct me if these structures are required. Will consider the possible options.

Reply:
DID Document metadata is a requirement of the DID Core specification. If we don't have them, then we don't have a DIDDoc. See: https://www.w3.org/TR/did-core/#did-document-metadata

In the Cosmos Cash implementation, they did all of the ones that were "should" in the DID Core spec and left the "may" ones, but ideally we should have all fields in the spec: https://github.com/allinbits/cosmos-cash/blob/main/docs/Explanation/ADR/adr-004-did.md#did-document-metadata

Finally, on deactivated status, it doesn't stop the DID from being read, it just ensures that there's a flag that it's no longer in use, and that no further update operations can be carried out on it. I don't see this as a business/legal risk.

Reply:
Okay sorry I looked at this in the actual changelog now and I see your point. Yes this metadata is different than the other one. Let's consider and discuss though whether this metadata is superseded by others.